src/concurrent/channel.cr

```diff
@@ -258,7 +260,10 @@ class Channel::Unbuffered(T) < Channel(T)
 
     @value.tap do
       @has_value = false
-      Scheduler.enqueue @sender.not_nil!
+      if sender = @sender
```

Comment:
I think we should still assert `@sender.not_nil!` here.

```cr
Scheduler.enqueue @sender.not_nil!
@sender = nil
```

Reply:
ok! not a possible race-condition to set `@sender` to `nil` after enqueuing it though?

Reply:
Not yet. `Channel` isn't thread safe currently anyway (you can see clear `@senders` after we enqueue it).